### ISSUE TITLE
Optimization

### ISSUE THREAD

**Issue by ghost** [SUAI-Smart-Space-Team/smart-edge-control-software#44]
Where is the optimization flag? I understand that this already works with delays of 5 seconds, but do not interfere with optimizing the Assembly code, it will at least slightly reduce the time of executing this "wonderful" code.
https://github.com/SUAI-Smart-Space-Team/smart-edge-control-software/blob/a2b89fd6abdd8753807688b5330af7c9b2ba41a5/makefile#L2

**Comment by marcfedorow** [comment]
I agree that this code is a `pure virtual` trash but I think you could be a little bit (or even byte) more gentle :)
Also there is no reason to optimize code with delays like this.
Well, ok, probably it should be optimized with \O1 (code length) if it deploys on some lightweight device, but if so, everything should be re-writed with pure C, function pointers, char[] manipulation and no `printf`s.
The other way is to use some high-level high-level programming language (I mean C is low-level HLPL while it gives access to memory and other stuff) like python or smth. It could close all the problems with memory leaks, UB, pointer missing and other.
Anyway, I think that optimization is not required so this issue could be closed.
